src/main/java/org/sonarlint/intellij/trigger/SonarLintCheckinHandler.java

```diff
@@ -89,19 +90,27 @@ public ReturnResult beforeCheckin(@Nullable CommitExecutor executor, PairConsume
     SonarLintSubmitter submitter = SonarLintUtils.get(project, SonarLintSubmitter.class);
     // this will block EDT (modal)
     try {
-      submitter.submitFilesModal(affectedFiles, TriggerType.CHECK_IN);
+      AtomicBoolean error = new AtomicBoolean(false);
```

Comment:
the goal is just to have a mutable boolean that is effectively final, not to be thread safe.